### ISSUE TITLE
dockerfile-hadolint checker ignores warning and info messages

### ISSUE THREAD

**Issue by Nexion** [flycheck/flycheck#1931]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
dockerfile-hadolint checker shows only errors, ignoring warning and info level messages of `hadolint`. Running it directly in terminal or via `flycheck-compile` shows messages of all error levels.

## Steps to reproduce
1. [dockerfile-mode](https://github.com/spotify/dockerfile-mode) is a Emacs requirement.
2. https://github.com/hadolint/hadolint should be installed.
3. Create a test Dockerfile, e.g.:
```dockerfile
FROM debian
MAINTAINER some guy
RUN apt-get update && apt-get upgrade
WORKDIR /usr/src/app
```
4. Running `hadolint Dockerfile` generates all three different levels of warnings. But in Emacs buffer for Dockerfile only the error will appear. Same if you open Flycheck errors with `C-c ! l`. This doesn't happen with other checkers, e.g. while editing elisp files I can see all the warning levels.

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
Flycheck should render all of the hadolint warnings.

## Screenshots
![term](https://user-images.githubusercontent.com/712414/150244506-176c00ad-a328-49c7-9e3c-3f998b5f2522.jpg)
![emacs](https://user-images.githubusercontent.com/712414/150244890-8965e39e-c130-42be-8909-7f268f53f642.jpg)

## System configuration
```
Syntax checkers for buffer Dockerfile in dockerfile-mode:

First checker to run:

  dockerfile-hadolint
    - may enable: yes
    - executable: Found at /usr/bin/hadolint

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: 32snapshot
Emacs version:    29.0.50
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes
hadolint version 2.8.0
flycheck and dockerfile-mode installed from git master

**Comment by fgabolde** [comment]
Seems to be fixable by using

```elisp
:command ("hadolint" "--no-color" source)
```

instead of `"-"`.

**Comment by cpitclaudel** [comment]
Thanks for the report.  It looks like hadolint now prints `-` instead of `/dev/stdin`, which is why we only display errors (not arnings and infos).  Please try https://github.com/flycheck/flycheck/pull/1978

**Comment by Nexion** [comment]
@cpitclaudel all good now, thanks for the patch.
